### PR TITLE
Suppressing SMTP Smuggling / CVE-2023-51764

### PR DESCRIPTION
**Pull request by jenszo**
The CVE might only be 5.3 but in the light of the majority of attacks being based on phishing/social engineering and an almost endless combination possibilities when email servies are self-hosted, I rather have this mitigated.

Please consider this PR.

BR Jens

### CODE REVIEW

File: server/etc/e-smith/templates/etc/postfix/main.cf/30smtpd_data_restrictions [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#
+# 30smtpd_data_restrictions -- Suppressing SMTP Smuggling
+# 
+# Postfix 2.x will not receive patches. 
+# This is the workaround as per https://www.postfix.org/smtp-smuggling.html
+#
+smtpd_data_restrictions = reject_unauth_pipelining
```
Comment (DavidePrincipi) [inline server/etc/e-smith/templates/etc/postfix/main.cf/30smtpd_data_restrictions:7]: The `main.cf` obtained by the template expansion should already disable pipelining as `smtpd_client_restrictions` here

https://github.com/NethServer/nethserver-mail/blob/5136df5a1f11261663262608749b55751253fd47/common/etc/e-smith/templates/etc/postfix/main.cf/00template_vars#L18

Do you think adding line 7 is safer?

File: server/etc/e-smith/templates/etc/postfix/main.cf/30smtpd_data_restrictions [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+smtpd_discard_ehlo_keywords = chunking, silent-discard
```
Comment (DavidePrincipi) [inline server/etc/e-smith/templates/etc/postfix/main.cf/30smtpd_data_restrictions:8]: https://www.postfix.org/postconf.5.html#smtpd_discard_ehlo_keywords

A "discarded keyword" is simply hidden in the EHLO response or is it completely disabled/forbidden?

**Review by DavidePrincipi — COMMENTED** [review]
Thank you for the pull request!

Please help me to understand if NS7 configuration is already conforming to the "short term workaround" described at https://www.postfix.org/smtp-smuggling.html

**Comment by gsanchietti** [comment]
No response from the author: I'm closing this stale PR.

**Pull request closed without merging**